src/TokenTests.sol

```diff
@@ -0,0 +1,174 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity >=0.8.0;
+
+import "./TokenChecks.sol";
+
+abstract contract TokenTests is TokenChecks {
+    address internal _token_;
+    string internal _contractName_;
+    string internal _tokenName_;
+    string internal _symbol_;
+    string internal _version_ = "1";
+    uint8 internal _decimals_ = 18;
+
+    function assertVarsSet() internal {
```

Comment:
Can we add these assert as regular test/s? (like `test_MintBurn` for example).

Reply:
I find it a better devex to have these as internal calls within the token tests to allow devs to quickly identify an incorrect lib configuration, but I guess I'm also fine to remove these `assertVarsSet` calls from the `test_` functions and have them just be another test if you're annoyed by the repetition.

Reply:
ok